coordinator/src/node/liquidated_positions.rs

```diff
@@ -0,0 +1,533 @@
+
+    match trader_direction {
+        Direction::Short => {
+            // if the trader is short that means the coordinator is long, so we have to check the
+            // coordinators liquidation price on the bid price.
+            if let Some(bid) = best_current_price.bid {
+                // check if coordinator needs to get liquidated
+                let coordinator_liquidation_price =
+                    calculate_long_liquidation_price(coordinator_leverage, average_entry_price);
+                let margin_call = coordinator_liquidation_price
+                    + coordinator_liquidation_price * margin_call_percentage;
+                if bid <= margin_call {
+                    return Some(Direction::Long);
+                }
+            }
+
+            // if the trader is short that means the coordinator is long, so we have to check the
+            // traders liquidation price on the ask price.
+            if let Some(ask) = best_current_price.ask {
+                // check if trader needs to get liquidated
+                let trader_liquidation_price =
+                    calculate_short_liquidation_price(trader_leverage, average_entry_price);
+                let margin_call =
+                    trader_liquidation_price - trader_liquidation_price * margin_call_percentage;
+
+                if ask >= margin_call {
+                    return Some(Direction::Short);
+                }
+            }
+        }
+        Direction::Long => {
+            // if the trader is long that means the coordinator is short, so we have to check the
+            // traders liquidation price on the bid price.
+            if let Some(bid) = best_current_price.bid {
+                // check if trader needs to get liquidated
+                let trader_liquidation_price =
+                    calculate_long_liquidation_price(trader_leverage, average_entry_price);
+                let margin_call =
+                    trader_liquidation_price + trader_liquidation_price * margin_call_percentage;
+
+                if bid <= margin_call {
+                    return Some(Direction::Long);
+                }
+            }
+
+            // if the trader is long that means the coordinator is short, so we have to check the
+            // coordiantors liquidation price on the ask price.
+            if let Some(ask) = best_current_price.ask {
+                // check if coordinator needs to get liquidated
+                let coordinator_liquidation_price =
+                    calculate_short_liquidation_price(coordinator_leverage, average_entry_price);
+                let margin_call = coordinator_liquidation_price
+                    - coordinator_liquidation_price * margin_call_percentage;
+
+                if ask >= margin_call {
+                    return Some(Direction::Short);
+                }
+            }
+        }
+    }
+
+    None
+}
+
+#[cfg(test)]
+mod tests {
```

Comment:
🔧 We should avoid "Test-Driven Design Ossification"[^1].

In summary, we should not right tens of tests against the same API and instead call a data-driven test API that delegates to the API we want to test. I've used the `check` idiom [here](https://github.com/get10101/10101/pull/2352/files#diff-dc08eded476b7979ce9fa96345782e5d1c570f176643482d8baf65678b9e4477R947), for example.

[^1]: https://matklad.github.io/2021/05/31/how-to-test.html#Test-Driven-Design-Ossification.

Reply:
This approach would have paid off twice before this PR even got merged lol.

Reply:
The tests have been removed as after the refactoring they were basically only testing `>=` and `<=`.